addons/binding/org.openhab.binding.upb/src/main/java/org/openhab/binding/upb/internal/UPBReader.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.upb.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class monitors the input stream of a UPB modem. This is done
+ * asynchronously. When messages are received, they are broadcast to all
+ * subscribed {@link Listener listeners}.
+ *
+ * @author Chris Van Orman
+ * @since 1.9.0
+ */
+public class UPBReader implements Runnable {
+
+    /**
+     * Listener class for handling received messages. A listener can be added by
+     * calling {@link UPBReader#addListener(Listener)}.
+     *
+     * @author cvanorman
+     *
+     */
+    public interface Listener {
+
+        /**
+         * Called whenever a message has been received from the UPB modem.
+         *
+         * @param message
+         *            the message that was received.
+         */
+        void messageReceived(UPBMessage message);
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(UPBReader.class);
```

Comment:
Logger should NOT be static according to the guidelines. [[source](http://docs.openhab.org/developers/development/guidelines.html#e-logging)]

Can you check all of them?